src/main/java/org/jboss/logmanager/LogContext.java

```diff
@@ -529,18 +496,13 @@ public void close() throws Exception {
             for (AutoCloseable handler : closeHandlers) {
                 handler.close();
             }
-            synchronized (this) {
-                attachmentKey1 = null;
-                attachmentKey2 = null;
-                final var value1 = attachmentValue1;
-                attachmentValue1 = null;
-                if (value1 instanceof AutoCloseable) {
-                    ((AutoCloseable) value1).close();
-                }
-                final var value2 = attachmentValue2;
-                attachmentValue2 = null;
-                if (value2 instanceof AutoCloseable) {
-                    ((AutoCloseable) value2).close();
+            final Map<?, ?> oldAttachments = (Map<?, ?>) attachmentHandle.getAndSet(this, Map.of());
```

Comment:
I should clarify that you don't need _any_ cast here :)

Reply:
The `VarHandle.getAndSet()` returns an `Object`, how do you get around casting?

Reply:
You're right, I forgot about it being signature-polymorphic (reviewed too fast).